lib/internal/http.js

```diff
@@ -32,6 +37,15 @@ function resetCache() {
   utcCache = undefined;
 }
 
+function validatePath(path) {
+  const pathRegex = /^[_0-9A-Za-z]+(\.[_0-9A-Za-z]+)*\.?:([1-9]|[1-9][0-9]{1,2}|[1-5][0-9]{3}|6[0-4][0-9]{3}|65[0-4][0-9]{2}|655[0-2][0-9]|6553[0-5])$/;
+  if (!pathRegex.test(path)) {
+    throw new ERR_INVALID_ARG_VALUE('options.path',
+                                    path, 'must be a valid host:port combo');
+  }
+  return path;
```

Comment:
I'd change this to simply return true/false and have the caller throw the exception (and rename to e.g. `isValidCONNECTPath()`.)

Rule of thumb: exceptions should be thrown close to the edge, where "edge" means the cross-over point between user code and core code.